File: archive-romhandbook-ui/lib/types/Mount.ts
export type Mount = {

    id: string

    name: string

    detail_url: string

    image: string | null

    description: string | null

    quality: string | null

    effect_text: string | null

    unlock_text: string | null

    jobs: string | null
}

export type MountDetail = {

    id: string

    detail_url: string

    image: string | null

    name: string

    mount_type: string | null

    description: string | null

    quality: string | null

    effect_text: string | null

    unlock_text: string | null

    jobs: string | null

    raw_html: string | null
}

export type ApiResponse<T> = {

    success: boolean

    data: T

    meta: unknown
}

export type PaginatedApiResponse<T> = {

    success: boolean

    data: T[]

    meta: {
        page: number
        limit: number
        total: number
        has_next: boolean
    }
}

File: backend-api/internal/models/mount.go
package models

type Mount struct {
	ID string `json:"id"`

	Name string `json:"name"`

	DetailURL string `json:"detail_url"`

	Image *string `json:"image"`

	Description *string `json:"description"`

	Quality *string `json:"quality"`

	EffectText *string `json:"effect_text"`

	UnlockText *string `json:"unlock_text"`

	Jobs *string `json:"jobs"`
}

type MountDetail struct {
	ID string `json:"id"`

	DetailURL string `json:"detail_url"`

	Image *string `json:"image"`

	Name string `json:"name"`

	MountType *string `json:"mount_type"`

	Description *string `json:"description"`

	Quality *string `json:"quality"`

	EffectText *string `json:"effect_text"`

	UnlockText *string `json:"unlock_text"`

	Jobs *string `json:"jobs"`

	RawHTML *string `json:"raw_html"`
}
